### PR TITLE
visual update of Demarche Selection

### PR DESCRIPTION
**Pull request by XavierJp**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -421,6 +421,8 @@ input[type='radio'][disabled] {
 /*fix: select makes container to grow*/
 select {
   width: 100%;
```
Comment (XavierJp) [inline src/App.css:423]: This fix the fact that the down arrow doesnot display on chrome as it is hidden by templete.dat.gouv.

I added the fix here as there was already a fix of template.data.gouv's CSS. I should have added a comment.

I tried to fix the select 100% width broken behaviour but didnot succeed. 

I think there should be a dedicated dropdown component in atoms. But I believe this PR is not the right place for it.

File: src/App.css [hunk 2]
```diff
@@ -516,3 +518,21 @@ section.full-width-section {
     width: inherit;
   }
 }
+
+/* generic flex declarative class */
```
Comment (XavierJp) [inline src/App.css:522]: Basic Declarative CSS class for ubiquitous flex layout

File: src/components/Form/DemarcheSelectionNotification.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import Loader from '../atoms/Loader';
+import { isEmpty } from 'lodash';
+import OpenInNewIcon from '../icons/open-in-new';
+
+const DemarcheSelectionNotification = ({
```
Comment (XavierJp) [inline src/components/Form/DemarcheSelectionNotification.js:6]: Not sure if this is convention but I put this notification component next to CopyNotification and HasNextNotification

File: src/components/atoms/Loader/index.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import './style.css';
```
Comment (XavierJp) [inline src/components/atoms/Loader/index.js:2]: I decided to redeclare a `Loader` component here as the `spinner` one cannot be used as is.  It needs a div wrapper with fixed height for instance, to display correctly. 

I believe we should always avoid such components that need external div wrapper in order to display correctly. At least this div wrapper should be included in the component and parametrized with props or higher order compoenents

File: src/components/atoms/Loader/style.css [hunk 5]
```diff
@@ -0,0 +1,33 @@
+.loader-container {
+  width: 100%;
+  margin: 20px auto;
+}
+
```
Comment (XavierJp) [inline src/components/atoms/Loader/style.css:5]: jumped lines oh yeah!!

File: src/pages/Login/index.css [hunk 6]
```diff
@@ -53,8 +53,8 @@
 .timeline > div > .time-indication {
   font-style: italic;
   position: absolute;
-  width: 80px;
-  right: -40px;
+  width: 90px;
```
Comment (XavierJp) [inline src/pages/Login/index.css:56]: Two lines issue fix for #151

File: src/components/atoms/Loader/style.css [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+.loader-container > .message {
+  margin-bottom: 10px;
+  font-style: italic;
+}
+
+.loader-container .loader {
+  border: 7px solid #c9d3df; /* Light grey */
+  border-top-color: #53657d; /* Blue */
```
Comment (XavierJp) [inline src/components/atoms/Loader/style.css:13]: delete comment

File: src/components/form-sections/DemarcheSection/index.js [hunk 8]
```diff
@@ -4,20 +4,20 @@ import { isEmpty, merge } from 'lodash';
 import { FormContext } from '../../Form';
 import { ScrollablePanel } from '../../Scrollable';
 import './index.css';
-import OpenInNewIcon from '../../icons/open-in-new';
+import DemarcheSelectionNotification from '../../Form/DemarcheSelectionNotification';
```
Comment (rdubigny) [inline src/components/form-sections/DemarcheSection/index.js:7]: I do not think this is the right place for this file.

File: src/components/form-sections/DemarcheSection/index.js [hunk 9]
```diff
@@ -4,20 +4,20 @@ import { isEmpty, merge } from 'lodash';
...
-  const [showInfo, setShowInfo] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
 
-  useEffect(
-    () => setShowInfo(selectedDemarcheId && selectedDemarcheId !== 'default'),
-    [selectedDemarcheId]
-  );
+  useEffect(() => {
+    setIsLoading(true);
+    window.setTimeout(() => setIsLoading(false), 900);
```
Comment (rdubigny) [inline src/components/form-sections/DemarcheSection/index.js:19]: that might not be the best way to handle setTimeout in React

File: src/App.css [hunk 10]
```diff
@@ -421,6 +421,8 @@ input[type='radio'][disabled] {
 /*fix: select makes container to grow*/
 select {
   width: 100%;
+  appearance: auto !important;
```
Comment (rdubigny) [inline src/App.css:424]: this break admin view

image

**Review by rdubigny — COMMENTED** [review]
It is quite careful to add comment in github but I rather not use this in the future because the point here is to validate the readability of the code without the comments.

File: src/components/form-sections/DemarcheSection/index.js (hunk 9 above)
Comment (rdubigny) [reply src/components/form-sections/DemarcheSection/index.js:19]: https://reactgo.com/settimeout-in-react-hooks/#clearing-settimeout

File: src/App.css (hunk 10 above)
Comment (rdubigny) [reply src/App.css:415]: avant : 

image

après : 

image

**Comment by rdubigny** [comment]
[Améliorer le visuel de la selection des cas d'usages](https://trello.com/c/njqZEK2q/825-am%C3%A9liorer-le-visuel-de-la-selection-des-cas-dusages)

**Review by rdubigny — APPROVED** [review]

**Pull request merged**